File: Modulo09/TimeJogador-Java/Time.java
public class Time {
    private String nome;
    private Jogador elenco[];

    public Time(String nome) {
        this.nome = nome;
        System.out.println("Criando time " + nome);
        elenco = new Jogador[22];
    }

    String getNome() {
        return this.nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public void adicionaJogador(int numero, Jogador jogador) {
        elenco[numero - 1] = jogador;
        jogador.setTime(this);
    }

    public void removeJogador(int numero) {
        elenco[numero - 1].setTime(null);
        elenco[numero - 1] = null;
    }

    public void imprime() {
        System.out.println("Elenco do " + nome);
        for (int i = 0; i < 22; i++) {
            if (elenco[i] != null) {
                System.out.println("(" + (i + 1) + ") ");
                elenco[i].imprime();
            }
        }
    }
}

File: Modulo06/TimeJogador-v2/Time.cpp
#include "Time.h"
#include <iostream>
#include <cstring>
using namespace std;

Time::Time(string nome) :
nome(nome) {
    cout << "Criando time " << nome << endl;
    this->elenco.reserve(22);
}

Time::~Time() {
    cout << "Destruindo time " << nome << endl;
}

string Time::getNome() const {
    return nome;
}

void Time::setNome(string nome) {
    this->nome = nome;
}

void Time::adicionaJogador(int numero, Jogador* jogador) {
    elenco[numero - 1] = jogador;
    jogador->setTime(this);
}

void Time::removeJogador(int numero) {
    elenco[numero - 1]->setTime(NULL);
    elenco[numero - 1] = NULL;
}

void Time::imprime() const {
    cout << "Elenco do " << nome << endl;
    for (int i = 0; i < 22; i++) {
        if (elenco[i] != NULL) {
            cout << "(" << (i + 1) << ") ";
            elenco[i]->imprime();
        }
    }
}

